src/plugins/dashboard/public/application/embeddable/grid/_dashboard_grid.scss

```
@@ -41,6 +41,16 @@
  * .dshLayout-withoutMargins only affects the panel styles themselves, see ../panel
  */
 
+/**
+ * 1. Fixes fullscreen mode for saved searches. Otherwise, sibling grids can obscure the grid in fullscreen mode.
+ * "euiDataGrid__restrictBody" is set to body element when a grid is in fullscreen
+ * https://github.com/elastic/kibana/issues/134032#issuecomment-1178091867
+ * It can be removed if we don't use inline styles (z-index, position) for Dashboard panels (react-grid-item)
+ */
+.euiDataGrid__restrictBody .dshDashboardGrid__item {
+  position: static !important; /* 1 */
```

It looks like mobile / small screens already has a `position: static !important` set, interestingly enough. It's not 100% clear to me why it doesn't 'just work' in that case, but as I was tinkering around I found a z-index unset that appears to work for both mobile and desktop:

```suggestion
.euiDataGrid__restrictBody .embPanel .embPanel__content {
  z-index: unset !important; /* 1 */
```

Unsetting z-index is probably a safer bet than position in any case. Let me know if that works for you!

---

@constancecchen Yes, this works great for desktop and mobile! Thank you very much for solving the issue 🎉